docs/dev/installer.md

```diff
@@ -0,0 +1,188 @@
+
+```bash
+./scripts/installer install --openshift
+```
+
+If you want a read only Dashboard, add the `--read-only` option to the command line:
+
+```bash
+./scripts/installer install --openshift --read-only
+```
+
+### OpenShift with Tekton Pipelines and Triggers installed manually using YAML manifests
+
+To install the Tekton Dashboard on OpenShift after installing Tekton Pipelines and Triggers manually using YAML manifests, you will need to override Tekton Pipelines and Triggers install namespaces.
+
+When installing from the manifests, Tekton Pipelines and Triggeers will be deployed in the `tekton-pipelines` namespace, whereas OpenShift Pipelines Operator uses the `openshift-pipelines` namespace.
+
+Therefore, you will need to add the `--pipelines-namespace tekton-pipelines` and `--triggers-namespace tekton-pipelines` options when calling the installer script:
+
+```bash
+./scripts/installer install --openshift --pipelines-namespace tekton-pipelines --triggers-namespace tekton-pipelines
+```
+
+If you want a read only Dashboard, add the `--read-only` option to the command line:
+
+```bash
+./scripts/installer install --openshift --pipelines-namespace tekton-pipelines --triggers-namespace tekton-pipelines --read-only
+```
+
+### Installing in a custom namespace
+
+You can install the Dashboard in the namespace of your choice, this works whatever the platform (Kubernetes / OpenShift) and install method (OpenShift Pipelines Operator / manifests).
+
+To tell the `installer` script the target namespace of your choice, add the `--namespace` option:
+
+```bash
+CUSTOM_NAMESPACE=my-namespace
+
+# for kubernetes
+./scripts/installer install --namespace $CUSTOM_NAMESPACE
+
+# for openshift / openshift pipelines operator
+./scripts/installer install --openshift --namespace $CUSTOM_NAMESPACE
+
+# for openshift / manifests
+./scripts/installer install --openshift --pipelines-namespace tekton-pipelines --triggers-namespace tekton-pipelines --namespace $CUSTOM_NAMESPACE
+```
+
+Add the `--read-only` to the command line that suits your platform and install method to deploy a read only Dashboard in your custom namespace.
+
+### Install ingress
+
+The `installer` script can create an ingress for your Dashboard service, to enable ingress creation add `--ingress-url` and optionally `--ingress-secret` if the ingress needs TLS support.
+
+```bash
+./scripts/installer install --ingress-url my-dashboard.my-domain.com --ingress-secret my-tls-certificate
+```
+
+## Uninstall command
+
+To uninstall the Dashboard, use the `uninstall` instead of the `install` command.
+
+**IMPORTANT NOTE:** this is VERY important when you uninstall using the `installer` script that you provide the same options that were given to the script at install time. This is needed for the script to know what was created at install time and must be deleted by the `uninstall` command.
```

Comment:
What are some possible consequences of uninstalling with the wrong options? e.g. leftover RBAC that might conflict with a future install. I think it would be useful to provide one or two examples so it's clear why this is important and also the types of problem to look out for to help troubleshooting if someone does forget to use the correct options.

Reply:
yeah, rbac, ingress, trying to uninstall in the wrong namespace, ...

Reply:
fixed